src/server/src/modules/user.ts

```diff
@@ -48,8 +51,32 @@ export const createUser = (
         firstName,
         lastName,
         role,
+        emailVerified: false
     });
-    return userToInsert.save();
+
+    await userToInsert.save();
+
+    let verificationKeyString: string;
+
+    do {
+        verificationKeyString = makeRandomString(/*length=*/20);
+    } while (await VerificationKey.findOne({ verificationKey: verificationKeyString }));
+
+    const verificationKey = VerificationKey.create({
+        user: userToInsert,
+        verificationKey: verificationKeyString
+    });
+
+    await verificationKey.save();
+    
+    // Send email with the verification link.
+    sendEmail(
+        email,
+        `Verify your email address`,
+        `Welcome to R'match! Please verify your email address so we know that it's you. http://localhost:3000/verify/${verificationKeyString}`,
```

Comment:
This is fine for development; however, in production, the link should start with https://obscure-ocean-12960.herokuapp.com/. So, the content of the link will depend on this value `process.env.NODE_ENV` if that value equals to "development" or "test" use localhost, if it equals to "production" use the heroku link

Reply:
Done.